waddrmgr/scoped_manager.go

```diff
@@ -2084,15 +2086,17 @@ func (s *ScopedKeyManager) ImportWitnessScript(ns walletdb.ReadWriteBucket,
 	script []byte, bs *BlockStamp, witnessVersion byte,
 	isSecretScript bool) (ManagedScriptAddress, error) {
 
+	digest := sha256.Sum256(script)
 	return s.importScriptAddress(
-		ns, script, bs, WitnessScript, witnessVersion, isSecretScript,
+		ns, digest[:], script, bs, WitnessScript, witnessVersion,
+		isSecretScript,
 	)
 }
 
 // importScriptAddress imports a new pay-to-script or pay-to-witness-script
 // address.
 func (s *ScopedKeyManager) importScriptAddress(ns walletdb.ReadWriteBucket,
-	script []byte, bs *BlockStamp, addrType AddressType,
+	scriptIdent []byte, script []byte, bs *BlockStamp, addrType AddressType,
```

Comment:
Alternatively, this could be a closure type that takes the script and returns an identifier. Then we can have pre-made versions for p2sh, and p2wsh, etc.

Reply:
Not sure if this was what you had in mind, but I added something to that effect with `Identity`. Can't really be more generic since the input isn't the same in all cases (raw script bytes vs. `*TapScript` struct or pub key).